src/Nethermind/Nethermind.Mev/Source/BundlePool.cs

```diff
@@ -141,6 +160,52 @@ public bool AddBundle(MevBundle bundle)
             return false;
         }
 
+        public bool AddMegabundle(MevMegabundle megabundle)
+        {
+            Metrics.MegabundlesReceived++;
+            BundleEventArgs bundleEventArgs = new(megabundle, megabundle.RelaySignature);
+            NewReceived?.Invoke(this, bundleEventArgs);
+
+            if (ValidateBundle(megabundle))
+            {
+                Address relayAddress = _ecdsa.RecoverAddress(megabundle.RelaySignature, megabundle.Hash)!;
+                if (IsTrustedRelay(relayAddress))
+                {
+                    Metrics.ValidMegabundlesReceived++;
+                    NewPending?.Invoke(this, bundleEventArgs);
+
+                    _megabundles.AddOrUpdate(relayAddress,
+                        _ => megabundle,
+                        (_, _) =>
+                        {
+                            RemoveSimulation(megabundle);
+                            return megabundle;
+                        });
+
+                    if (megabundle.BlockNumber == HeadNumber + 1)
+                    {
+                        TrySimulateBundle(megabundle);
+                    }
+
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private bool BundleInTimestampRange(MevBundle bundle, UInt256 minTimestamp, UInt256 maxTimestamp)
+        {
+            bool bundleIsInFuture = bundle.MinTimestamp != UInt256.Zero && minTimestamp < bundle.MinTimestamp;
+            bool bundleIsTooOld = bundle.MaxTimestamp != UInt256.Zero && maxTimestamp > bundle.MaxTimestamp;
+            return !bundleIsInFuture && !bundleIsTooOld;
+        }
+        
+        private bool IsTrustedRelay(Address relayAddress)
+        {
+            return _mevConfig.GetTrustedRelayAddresses().Contains(relayAddress);
```

Comment:
every time this is called GetTrustedRelayAddresses parses the string, please save it to some HashSet in constructor.